src/SearchName.js

```diff
@@ -0,0 +1,66 @@
+import axios from 'axios';
+import { useEffect, useState } from 'react';
+import ResultsName from './ResultsName';
+
+const SearchName = () => {
+
+  const [ searchValue, setSearchValue ] = useState('');
+  const [ input, setInput ] = useState('');
+  const [ results, setResults ] = useState([]);
+  const [ error, setError ] = useState(null);
+
+    useEffect(() => {
+        axios({
+          url:'https://www.thecocktaildb.com/api/json/v1/1/search.php?',
+          params:{
+            s:searchValue,
+          }
+        }).then( (cocktails) => {
+          if (cocktails.data.drinks === null){
+            throw Error ('No cocktails match your search term, please try again');
+          }
+          setResults(cocktails.data.drinks);
+        }).catch(error => {
+            setError(error.message);
+        });
+      }, [input]);
+
+    const handleClick = (e) => {
+      e.preventDefault();
+      setInput(searchValue);
+    }
+    const handleOnChange = (e) => {
+      setSearchValue(e.target.value);
+    }
+
+    return(
+        <div>
+          <section className='searchBar'>
+            <h3>Search by Cocktail Name</h3>
+            <form>
+              <label htmlFor='searchName' className='sr-only'>Enter Name</label>
+              <input 
+              type='text'
+              name='searchName'
+              placeholder='Enter Name'
+              onChange={handleOnChange}
+              />
+              <button type='submit' onClick={handleClick}>Search</button>
```

Comment:
The onClick event is better to be linked with form element (replaced with onSubmit), like in Javascript, we always add event listener to form rather than button, I think?